src/settings.js

```diff
@@ -382,6 +382,10 @@ var NO_FILESYSTEM = 0; // If set, does not build in any filesystem support. Usef
 var FORCE_FILESYSTEM = 0; // Makes full filesystem support be included, even if statically it looks like it is not
                           // used. For example, if your C code uses no files, but you include some JS that does,
                           // you might need this.
+var NODERAWFS = 0; // This dynamically disables virtual filesystem when the build result runs on Node.js
```

Comment:
perhaps this should just make the build node.js-only, that seems simpler? is there a benefit to letting it run on non-node with different behavior?

Reply:
NODERAWFS only replaces low-level APIs, otherwise it depends on original FS methods including `FS.read/writeFile`. `FS.ensureErrnoError`, `FS.nextfd`, etc. I think there is no value to make it node.js-only when we still depend on original FS.

Reply:
>which shell you are in should not matter

Yeah... it can be confusing in other shell environments. We may make it throw first and see what's going on.

Reply:
Updated but probably we cannot run a failure test on CI.

Reply:
Thanks, and yeah, makes sense about testing.

This PR looks good to me now, just one last thing, please update the comment here in settings.js to say it will throw if not in Node.js.